docs/reference/contributing/guidelines/workflow.md

```diff
@@ -28,6 +28,14 @@ A member of the Mbed team must verify bug fixes before we pull the fixes into th
 
 The last line in your commit message description should say “Fixes #deadbeef”, where “deadbeef” is the issue number in GitHub. This allows GitHub to automatically close the issue when the commit is merged into the default branch.
 
+#### Features
+
+Any feature should be implemented on the branch in mbed OS repository first. 
```

Comment:
New features should initially be implemented on a separate branch in the mbed OS repository. The naming convention for a feature branch is: "feature-<feature name>"